### PR TITLE
Update pull_request_template.md

### PR DESCRIPTION
**Pull request by lesliecdubs**
Adding more depth to the standard PRC PR template to ensure library maintainers and contributors have provided enough context in order to confidently review changes before merge.

### CODE REVIEW

**Review by broccolinisoup — APPROVED** [review]
Looks great! Thank you for this PR.

**Comment by lesliecdubs** [comment]
Thanks for the ideas @joshblack. I'd love for the changes we decide on for these templates to be instated across both PRC and PVC to keep the contribution experience more seamless between libraries, and since the content is generic enough to be relevant across both. That said, I'm cc'ing the PVC folks here in case they would like to weigh in too: @primer/rails-reviewers

My initial thoughts:

> Have Closes # at the top of the issue

I think this is analogous to the new "List the issues that this change affects" heading. I moved that up to the top in [this commit](https://github.com/primer/react/pull/3344/commits/9053cf2f11ead572034dc4293a1577bfa7cb0419), but feel free to suggest further changes directly on this PR.

> Consolidate certain sections into a general "about" section, specifically:
> What are you trying to accomplish
> What approach did you choose and why

This is okay with me assuming it doesn't mean we'll get paragraphs of info in that general section. @joshblack can you drop a suggestion into this PR with the change you're proposing?

> Remove Screenshots as a top-level heading and roll it into About above

@joshblack can you add a suggestion for this change, and explicitly mention "screenshots" in instructions in the "about" section? I think it's fine for PRs to go up without a screenshot if they don't significantly impact UI, but I would still expect screenshots to be added for any relevant changes.

> Add an explicit Changelog section that details the changes the PR makes from the perspective of an end user, e.g. what has been added, changed, or removed

Is this for the benefit of reuse in our release notes, or for the benefit of the reviewer? Either way, seems like a reasonable addition to me.

> Add a "Testing & Reviewing" section
> This provides a way to provide reviewers with specific context or things to pay attention to when reviewing
> This could also include Merge checklist
> Remove the Accessibility section, I'm not really sure how to apply this to the work that we have and what it would add given that this work is part of our practices

I'm not too precious about the current heading structure in the template, so if you'd like to propose changes please feel free. That said, I think it's important that we maintain:
-  a merge checklist of key actions that must be done/true before a reviewer will merge
- a callout for accessibility, whether this is that automatic axe checks should pass in CI or that any UI work has been tested manually with a screenreader and/or Accessibility Insights tools

File: .github/pull_request_template.md
Comment (joshblack) [inline .github/pull_request_template.md:16]:
```suggestion
```

File: .github/pull_request_template.md
Comment (joshblack) [inline .github/pull_request_template.md:13]:
```suggestion
<!-- Provide the GitHub issue that this issue closes. Start typing the number or name of the issue after the # below -->

Closes #

<!-- Provide an overview of the changes, including any screenshots when helpful -->
```

File: .github/pull_request_template.md
Comment (joshblack) [inline .github/pull_request_template.md:34]: I think this section could be framed as "testing & reviewing" in order to provide a way for authors to provide reviewers with specific context into the PR and how it could be tested/reviewed

```suggestion
## Testing & Reviewing

<!-- Describe any specific details to help reviewers test or review this Pull Request -->
```

File: .github/pull_request_template.md
Comment (joshblack) [inline .github/pull_request_template.md:23]: I think this section is challenging from the library side as it's hard or limiting to express changes in terms of:

- Observability
- Availability
- The ability of a change to be rolled back
- The impact of a change to SLOs

Instead of evaluating the risk of a change as it relates to a production environment, we could talk about the types of changes that the PR introduces from the perspective of the end user.

```suggestion
## Changelog

<!-- List out API changes that this Pull Request introduces -->

### New

### Changed

### Removed
```

File: .github/pull_request_template.md
Comment (joshblack) [inline .github/pull_request_template.md:29]: This seems like something that can be rolled into the overview section above

```suggestion
```

File: .github/pull_request_template.md
Comment (joshblack) [inline .github/pull_request_template.md:43]: This section feels like a "practice" detail in that accessibility is a part of the work we do and that work must not introduce a11y/axe violations.

For example, in what scenario would this section be filled out with "New axe violation" selected as an option? It seems like this would mostly be "No new axe scan violations" and occasionally "Fixes axe scan violations" but this detail could be included in the changelog section above

**Review by joshblack — COMMENTED** [review]
Tried to leave review comments, hope that they make sense 🤔 This kind of context can make it challenging to request changes as they might not make sense in isolation.

File: .github/pull_request_template.md
Comment (lesliecdubs) [reply .github/pull_request_template.md:16]: Can you provide any more context for this suggestion? The instructions note that this section is fine to remove if the changes do not impact UI.

Comment (lesliecdubs) [reply .github/pull_request_template.md:16]: I see now that you moved the screenshot suggestion to elsewhere in the template. That's fine with me; I'll push a recommended change to the instruction there but otherwise I'm good with this.

File: .github/pull_request_template.md
Comment (lesliecdubs) [reply .github/pull_request_template.md]: Resolved in commit https://github.com/primer/react/pull/3344/commits/6631c1b2ea7cfe9226201b606ef0261ba57ab1b8

File: .github/pull_request_template.md
Comment (lesliecdubs) [reply .github/pull_request_template.md]: Updated, see diff: https://github.com/primer/react/pull/3344/files#diff-b2496e80299b8c3150b1944450bd81c622e04e13d15c411d291db0927d75fd6bR21

File: .github/pull_request_template.md
Comment (lesliecdubs) [reply .github/pull_request_template.md]: Adopted the recommended change and added a section for "rollout strategy": https://github.com/primer/react/pull/3344/commits/62bedc473eff54e11c46cd0a4794e0d9dcb7d310

File: .github/pull_request_template.md
Comment (lesliecdubs) [reply .github/pull_request_template.md]: Removed this section in https://github.com/primer/react/pull/3344/commits/8fb13a10eb835d35835123fbcefa12a572c96e8c

File: .github/pull_request_template.md
Comment (lesliecdubs) [reply .github/pull_request_template.md:33]: Are we running axe scans in CI in `primer/react` yet, or is that still tbd? If the answer is yes, I think we can remove this whole section. 

If not, I'd probably prefer to maintain a section dedicated to accessibility for now to prompt contributors to ensure compliance when opening the PR.

**Comment by lesliecdubs** [comment]
Thanks for the feedback @joshblack! I iterated a bit more based on your thoughts and subsequent conversations. Would love your 👀s again! 

Also added a review request for other key PRC maintainers before we merge.

File: .github/pull_request_template.md
Comment (joshblack) [inline .github/pull_request_template.md:1]:
```suggestion
```

File: .github/pull_request_template.md
Comment (joshblack) [inline .github/pull_request_template.md:8]:
```suggestion
### Changelog
```

File: .github/pull_request_template.md
Comment (joshblack) [inline .github/pull_request_template.md:10]: Updated heading level based on above change and added comment for section

```suggestion
#### New

<!-- List of things added in this PR -->

#### Changed

<!-- List of things changed in this PR -->

#### Removed

<!-- List of things removed in this PR -->
```

File: .github/pull_request_template.md
Comment (joshblack) [reply .github/pull_request_template.md:33]: > Are we running axe scans in CI in primer/react yet, or is that still tbd? If the answer is yes, I think we can remove this whole section.

@lesliecdubs it should be enabled and required at this point 👍

**Review by joshblack — COMMENTED** [review]
Left a couple suggestions for hierarchy and a couple comments that could be useful, thanks for putting this all together!

**Comment by joshblack** [comment]
> Also not sure how to clear this [failing status check](https://github.com/primer/react/actions/runs/6125661359/job/16628613525?pr=3344#step:5:19); any tips?

@lesliecdubs I think running `npm run format` should do the trick! Just needs to be formatted by prettier

**Review by siddharthkp — APPROVED** [review]

File: .github/pull_request_template.md
Comment (lesliecdubs) [reply .github/pull_request_template.md]: Thanks for confirming. I've removed this redundant section of the template in https://github.com/primer/react/pull/3344/commits/48815b5dd6bef7ab2b565183995c4cf0d9d8dea1.

**Review by joshblack — APPROVED** [review]

**Pull request merged**
